apps/extension/src/routes/page/onboarding/height.tsx

```diff
@@ -0,0 +1,73 @@
+import { BackIcon } from '@repo/ui/components/ui/icons/back-icon';
+import { Button } from '@repo/ui/components/ui/button';
+import {
+  Card,
+  CardContent,
+  CardDescription,
+  CardHeader,
+  CardTitle,
+} from '@repo/ui/components/ui/card';
+import { FadeTransition } from '@repo/ui/components/ui/fade-transition';
+import { usePageNav } from '../../../utils/navigate';
+import { PagePath } from '../paths';
+import { FormEvent, MouseEvent, useState } from 'react';
+import { Input } from '@repo/ui/components/ui/input';
+import { localExtStorage } from '../../../storage/local';
+
+export const ImportWalletCreationHeight = () => {
+  const navigate = usePageNav();
+  const [blockHeight, setBlockHeight] = useState('');
+
+  const handleSubmit = async (event: MouseEvent | FormEvent) => {
+    event.preventDefault();
+
+    if (blockHeight) {
+      // Save the block height to local extension storage
+      await localExtStorage.set('walletCreationBlockHeight', Number(blockHeight));
+    }
+
+    navigate(PagePath.SET_PASSWORD);
+  };
+
+  const handleClick = (event: MouseEvent | FormEvent) => {
+    handleSubmit(event).catch((error: unknown) => {
+      if (error instanceof Error) {
+        console.error('Error during submission:', error.message);
+      } else {
+        console.error('Unexpected error:', error);
+      }
+    });
+  };
```

Comment:
Think this logic in the component can live in zustand too. Maybe all this creation height stuff should be its own zustand slice

Reply:
created dedicated `zustand` slice and removed `handleClick` logic.